src/main.rs

```diff
@@ -3,10 +3,105 @@ use clap::{Arg, Command};
 use linkme::distributed_slice;
 use metriken::Lazy;
 use ringlog::*;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::SystemTime;
+use tokio::sync::RwLock;
 
 type Duration = clocksource::Duration<clocksource::Nanoseconds<u64>>;
 type Instant = clocksource::Instant<clocksource::Nanoseconds<u64>>;
 
+type HistogramSnapshots = HashMap<String, metriken::histogram::Snapshot>;
```

Comment:
Not for this changeset, but if this is the abstraction that both rpc-perf and rezolus end up using and it works well, it might be worth hoisting it into metriken.